python/lsst/validate/drp/validate.py

```diff
@@ -0,0 +1,368 @@
+#!/usr/bin/env python
+
+# LSST Data Management System
+# Copyright 2008-2016 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+
+from __future__ import print_function, absolute_import
+
+import numpy as np
+import yaml
+
+import lsst.afw.geom as afwGeom
+import lsst.afw.image as afwImage
+from lsst.afw.table import SourceCatalog, SchemaMapper, Field
+from lsst.afw.table import MultiMatch, SimpleRecord, GroupView
+import lsst.daf.persistence as dafPersist
+import lsst.pipe.base as pipeBase
+
+from .calcSrd import calcAM1, calcAM2
+from .check import checkAstrometry, checkPhotometry, positionRms
+from .print import printPA1, printPA2, printAM1, printAM2
+from .plot import plotAstrometry, plotPhotometry, plotPA1, plotAM1, plotAM2
+from .util import getCcdKeyName, repoNameToPrefix
+
+
+def loadDataIdsAndParameters(configFile):
+    """Load data IDs, magnitude range, and expected metrics from a yaml file.
+
+    Inputs
+    ------
+    configFile : str
+        YAML file that stores visit, filter, ccd,
+        good_mag_limit, medianAstromscatterRef, medianPhotoscatterRef, matchRef
+
+    Returns
+    -------
+    dict, float, float, float
+        dataIds, good_mag_limit, medianRef, matchRef
+    """
+    stream = open(configFile, mode='r')
+    data = yaml.load(stream)
+
+    ccdKeyName = getCcdKeyName(data)
+    visitDataIds = constructDataIds(data['visits'], data['filter'],
+                                    data[ccdKeyName], ccdKeyName)
+
+    return (visitDataIds,
+            data['good_mag_limit'],
+            data['medianAstromscatterRef'],
+            data['medianPhotoscatterRef'],
+            data['matchRef'],
+           )
+
+
+def constructDataIds(visits, filter, ccds, ccdKeyName=None):
+    """Construct a comprehensive set of Butler Data IDs for a repository.
```

Comment:
Rather: returns a data ID consisting of every combination of visit & ccd for each filter, right? There could be a lot more things in the repository that aren't described by IDs of this form. Also, this makes no attempt to check if the data is there (what if some CCDs don't exist for a given visit)?

As a minor nit, if the function is "for a given filter, give me all combinations of visits and ccds" it would be more logical to order the arguments `filter, visits, ccds`.